From 8f06b9e7b79801009de395cb98e221a0af1d1d7d Mon Sep 17 00:00:00 2001
From: Manikanta Mylavarapu <quic_mmanikan@quicinc.com>
Date: Wed, 27 Dec 2023 10:07:23 +0530
Subject: [PATCH 367/500] rproc: add non secure pil kconfig

Non-secure PIL is used to bring up internal q6 on emulation platforms.
So add config in Kconfig file to build non-secure PIL code in multipd
rproc driver.

Change-Id: Ib444f22784d778c4982751567809661ddb24660b
Signed-off-by: Manikanta Mylavarapu <quic_mmanikan@quicinc.com>
---
 drivers/remoteproc/Kconfig | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/remoteproc/Kconfig b/drivers/remoteproc/Kconfig
index 22f7b749e064..0426c06f1d5e 100644
--- a/drivers/remoteproc/Kconfig
+++ b/drivers/remoteproc/Kconfig
@@ -254,6 +254,15 @@ config QCOM_Q6V5_MPD
 	  for the Hexagon based MultiPD model remote processors on e.g. IPQ5018.
 	  This is trustZone wireless subsystem.
 
+config QCOM_NON_SECURE_PIL
+	tristate "Qualcomm Non secure Peripheral Image Loader"
+	depends on OF && ARCH_QCOM
+	depends on QCOM_Q6V5_MPD
+	help
+	  Say y here to support the Qualcomm Non Secure Peripheral Image Loader
+	  for the Hexagon based MultiPD model remote processors on emulation
+	  platforms.
+
 config QCOM_Q6V5_WCSS
 	tristate "Qualcomm Hexagon based WCSS Peripheral Image Loader"
 	depends on OF && ARCH_QCOM
-- 
2.34.1

